c++/src/capnp/serialize.h

```diff
@@ -169,7 +176,11 @@ void writeMessage(kj::OutputStream& output, kj::ArrayPtr<const kj::ArrayPtr<cons
 // Specializations for reading from / writing to file descriptors.
 
 class StreamFdMessageReader: private kj::FdInputStream, public InputStreamMessageReader {
-  // A MessageReader that reads from a stream-based file descriptor.
+  // A MessageReader that reads from a stream-based file descriptor. Example:
+  //
+  //     auto fs = kj::newDiskFilesystem();
+  //     auto file = fs->getCurrent().openFile(kj::Path::parse("mydata"));
+  //     capnp::StreamFdMessageReader capnpReader { KJ_ASSERT_NONNULL(file->getFd()) };
```

Comment:
This isn't a great demonstration of the KJ filesystem APIs since you're just pulling out the underlying FD. This won't actually even work on Windows, which doesn't use file descriptors for the filesystem.

An example using `open()` might be better:

```suggestion
  //     int fd = open("mydata", O_RDONLY);
  //     assert(fd >= 0);
  //     capnp::StreamFdMessageReader capnpReader(fd);
```

That said I'm not sure this is particularly exciting either, and it still has a bunch of weird issues on Windows: `open` and `O_RDONLY` both need to be prefixed with `_` on Windows, and `_O_BINARY` is really needed too, but other systems don't have that flag.

Plus, people should not be using `StreamFdMessageReader` for files anyway, they should be using mmap, like in your other example. The mmap approach is usually far superior. `StreamFdMessageReader` is really intended for sockets, pipes, etc.

So... I dunno, maybe there isn't a good example to show here.

Reply:
> That's a case where you can't use mmap.

Why not? Temp files are files.

(Also a newly-created temp file would be empty so I guess there's nothing to read.)

Maybe the example could open a named pipe? That's a case where you actually can't use mmap...